auth/auth_test.go

```diff
@@ -573,54 +587,171 @@ func TestFilterToAvailableSince(t *testing.T) {
+			assert.Equal(t, nil, auth.Save(role))
 
 			user, _ := auth.NewUser("testUser", "password", ch.SetOf("explicit1"))
 			user.setChannels(nil)
 			err := auth.Save(user)
-			goassert.Equals(t, err, nil)
+			assert.Equal(t, nil, err)
 
 			user2, err := auth.GetUser("testUser")
-			goassert.Equals(t, err, nil)
+			assert.Equal(t, nil, err)
 
 			channelsSinceStar, secondaryTriggersStar := user2.FilterToAvailableChannelsForSince(ch.SetOf("*"), sinceClock)
 			channelA_Star, ok := channelsSinceStar["A"]
 			log.Printf("channelA_Star: %s", channelA_Star)
-			goassert.True(t, ok)
-			goassert.True(t, channelA_Star.Equals(tc.expectedResult))
+			assert.True(t, ok)
+			assert.True(t, channelA_Star.Equals(tc.expectedResult))
 			secondaryTriggerStar, ok := secondaryTriggersStar["A"]
 			log.Printf("secondaryTrigger %v, expected %v", secondaryTriggerStar, tc.expectedSecondaryTrigger)
-			goassert.True(t, secondaryTriggerStar.Equals(tc.expectedSecondaryTrigger))
+			assert.True(t, secondaryTriggerStar.Equals(tc.expectedSecondaryTrigger))
 
 			channelsSince, secondaryTriggersSince := user2.FilterToAvailableChannelsForSince(ch.SetOf("A"), sinceClock)
 			channelA_Single, ok := channelsSince["A"]
 			log.Printf("channelA_Single: %s", channelA_Single)
-			goassert.True(t, ok)
-			goassert.True(t, channelA_Single.Equals(tc.expectedResult))
+			assert.True(t, ok)
+			assert.True(t, channelA_Single.Equals(tc.expectedResult))
 			secondaryTriggerSince, ok := secondaryTriggersSince["A"]
 			log.Printf("secondaryTrigger %v, expected %v", secondaryTriggerSince, tc.expectedSecondaryTrigger)
-			goassert.True(t, secondaryTriggerSince.Equals(tc.expectedSecondaryTrigger))
+			assert.True(t, secondaryTriggerSince.Equals(tc.expectedSecondaryTrigger))
 
 			channelBSince, secondaryTriggersB := user2.FilterToAvailableChannelsForSince(ch.SetOf("B"), sinceClock)
 			log.Printf("channelBSince: %s", channelBSince)
-			goassert.True(t, len(channelBSince) == 0)
-			goassert.True(t, len(secondaryTriggersB) == 0)
+			assert.True(t, len(channelBSince) == 0)
+			assert.True(t, len(secondaryTriggersB) == 0)
 
 			channelsSinceMulti, secondaryTriggersMulti := user2.FilterToAvailableChannelsForSince(ch.SetOf("A", "B"), sinceClock)
 			log.Printf("syncGrant1Multi: %s", channelsSinceMulti)
-			goassert.True(t, len(channelsSinceMulti) == 1)
+			assert.True(t, len(channelsSinceMulti) == 1)
 			channelA_Multi, ok := channelsSinceMulti["A"]
-			goassert.True(t, ok)
-			goassert.True(t, channelA_Multi.Equals(tc.expectedResult))
+			assert.True(t, ok)
+			assert.True(t, channelA_Multi.Equals(tc.expectedResult))
 			secondaryTriggerMulti, ok := secondaryTriggersMulti["A"]
 			log.Printf("secondaryTrigger %v, expected %v", secondaryTriggerMulti, tc.expectedSecondaryTrigger)
-			goassert.True(t, secondaryTriggerMulti.Equals(tc.expectedSecondaryTrigger))
+			assert.True(t, secondaryTriggerMulti.Equals(tc.expectedSecondaryTrigger))
 
 		})
 	}
 
 }
 
+func TestConcurrentUserWrites(t *testing.T) {
```

Comment:
What about running this in a loop in order to increase the chances of concurrent updates?

Reply:
This was consistently failing pre-fix, so I don't think that's needed.